### ISSUE TITLE
Test Cases are getting skipped when run selenium tests in parallel with error info as 'java.io.IOException: unexpected end of stream on Connection`

### ISSUE THREAD

**Issue by nagendra-devireddy** [zalando/zalenium#895]
## 🐛 Bug Report

Tests are getting  skipped  when I run tests in parallel and getting error message as "java.io.IOException: unexpected end of stream on Connection"

## To Reproduce

We are running zalenium in Rancher environment.  We have scheduled 250 test cases with MaxTestsessions 1 around 220 test cases are skipped with parallel 5 threads, updated MaxTestsessions 5 with parallel 5 threads 170 tests are skipped and updated MaxTestsessions 10 with parallel 5 threads 210 tests are skipped. 

Zalenium docker logs i found below error logs 
/11/2019 3:21:51 PMjava.io.IOException: unexpected end of stream on Connection{127.0.0.1:40032, proxy=DIRECT hostAddress=/127.0.0.1:40032 cipherSuite=none protocol=http/1.1}
3/11/2019 3:21:51 PM  at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:208)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:88)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
3/11/2019 3:21:51 PM  at org.openqa.selenium.remote.internal.OkHttpClient$Factory$1.lambda$createClient$1(OkHttpClient.java:152)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
3/11/2019 3:21:51 PM  at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
3/11/2019 3:21:51 PM  at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
3/11/2019 3:21:51 PM  at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
3/11/2019 3:21:51 PM  at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
3/11/2019 3:21:51 PM  at okhttp3.RealCall.execute(RealCall.java:77)
3/11/2019 3:21:51 PM  at org.openqa.selenium.remote.internal.OkHttpClient.execute(OkHttpClient.java:103)
3/11/2019 3:21:51 PM  at org.openqa.grid.internal.TestSession.sendRequestToNode(TestSession.java:422)
3/11/2019 3:21:51 PM  at org.openqa.grid.internal.TestSession.forward(TestSession.java:229)
3/11/2019 3:21:51 PM  at org.openqa.grid.web.servlet.handler.RequestHandler.forwardRequest(RequestHandler.java:99)
3/11/2019 3:21:51 PM  at org.openqa.grid.web.servlet.handler.RequestHandler.process(RequestHandler.java:133)
3/11/2019 3:21:51 PM  at org.openqa.grid.web.servlet.DriverServlet.process(DriverServlet.java:85)
3/11/2019 3:21:51 PM  at org.openqa.grid.web.servlet.DriverServlet.doGet(DriverServlet.java:63)
3/11/2019 3:21:51 PM  at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
3/11/2019 3:21:51 PM  at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.servlet.ServletHolder.handle(ServletHolder.java:865)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
3/11/2019 3:21:51 PM  at io.prometheus.client.filter.MetricsFilter.doFilter(MetricsFilter.java:170)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.servlet.ServletHandler.doHandle(ServletHandler.java:533)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.security.SecurityHandler.handle(SecurityHandler.java:548)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.servlet.ServletHandler.doScope(ServletHandler.java:473)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.session.SessionHandler.doScope(SessionHandler.java:1564)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ContextHandler.doScope(ContextHandler.java:1242)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.StatisticsHandler.handle(StatisticsHandler.java:174)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.Server.handle(Server.java:503)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.HttpChannel.handle(HttpChannel.java:364)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.server.HttpConnection.onFillable(HttpConnection.java:260)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.io.FillInterest.fillable(FillInterest.java:103)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
3/11/2019 3:21:51 PM  at org.seleniumhq.jetty9.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
3/11/2019 3:21:51 PM  at java.lang.Thread.run(Thread.java:748)
3/11/2019 3:21:51 PMCaused by: java.io.EOFException: \n not found: limit=0 content=…
3/11/2019 3:21:51 PM  at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:237)
3/11/2019 3:21:51 PM  at okhttp3.internal.http1.Http1Codec.readHeaderLine(Http1Codec.java:215)
3/11/2019 3:21:51 PM  at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
3/11/2019 3:21:51 PM  ... 61 common frames omitted 

Earlier i used to see "cannot forword the request unexpected end of stream on connection"
so i am passing Apache for ZALENIUM_EXTRA_JVM_PARAMS

here is the docker-compose what we are running 
version: '2'
services:  
  zalenium:
    privileged: true
    image: dosel/zalenium 
    network_mode: host
    environment:
    - PULL_SELENIUM_IMAGE=true
    - SELENIUM_CONTAINER_LABELS=io.rancher.container.network=true
    - SELENIUM_WAIT_FOR_CONTAINER=false    
    - ZALENIUM_EXTRA_JVM_PARAMS=-Dwebdriver.http.factory=apache
    - NEW_SESSION_WAIT_TIMEOUT=36000000
    dns_search:
    - xyz.com
    tty: true         
    labels:
      io.rancher.container.pull_image: always
      io.rancher.scheduler.affinity:host_label: zlab=1
      type: lb          
    volumes:
    - /var/run/docker.sock:/var/run/docker.sock 
    - /space/common:/home/seluser/videos
    command: >
          start --timeZone "Asia/Calcutta"
                  --maxTestSessions 10
                
    
## Expected behavior

Shouldn't skip the tests

## Environment

OS:  Linux
Zalenium Image Version(s):  Latest
Docker version:  Docker version 18.09.0, build 4d60db4
Docker-Compose version (if applicable):  2
Exact Docker command to start the containers (if using docker-compose, provide 
the docker-compose file as well):
If using Kubernetes, specify your environment, and if relevant your manifests: No

**Comment by diemol** [comment]
This is a duplicate of #742, would be great if you can help us over there to debug it and find the root cause.
